packages/@lwc/engine-core/src/framework/def.ts

```diff
@@ -174,6 +196,11 @@ function createComponentDef(Ctor: LightningElementConstructor): ComponentDef {
         renderMode = ctorRenderMode === 'light' ? RenderMode.Light : RenderMode.Shadow;
     }
 
+    let formAssociated = superDef.formAssociated;
+    if (!isUndefined(ctorFormAssociated)) {
+        formAssociated = ctorFormAssociated;
+    }
```

Comment:
```suggestion
    const formAssociated = Boolean(superDef.formAssociated);
```

[I looked into it](https://codepen.io/nolanlawson-the-selector/pen/poqZooZ?editors=1010), and it seems that the browser's treatment of `formAssociated` is based on truthiness. It would make our internal code simpler if we set `formAssociated` to either true/false in one place in the code, so that we could rely on it being a boolean everywhere else.

Reply:
Or `undefined | boolean` is also fine – just that we don't actually know what type `superDef.formAssociated` is right? The user could put `static formAssociated = 'yolo'`.